### PR TITLE
JSUI-2373 Add unit tests for facet values

### PR DESCRIPTION
**Pull request by ThibodeauJF**
Moved some files in sub folders
Mostly the new code is in MLFacetValueTest & MLFacetValueRendererTest

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

File: unitTests/ui/MLFacet/MLFacetValues/MLFacetValueTest.ts [hunk 1]
```diff
@@ -60,9 +66,37 @@ export function MLFacetValueTest() {
...
+
+    it(`when getting formattedCount with a zero value
+      it should return an empty string`, () => {
+      mLFacetValue.numberOfResults = 0;
+      expect(mLFacetValue.formattedCount).toBe('');
+    });
+
+    it(`when using the valueCaption option with a function
+      should bypass it and return the original value`, () => {
+      initializeComponent({ valueCaption: () => 'allo' });
```
Comment (samisayegh) [inline unitTests/ui/MLFacet/MLFacetValues/MLFacetValueTest.ts]: A better pattern would be to write your options to a variable on ~line 11, and then have `initializeComponent` read that variable. This allows you to sequentially build your options in nested `describe`/`it` blocks and keeps the `initalizeComponent()` line short. e.g.

```
let options: IMLFacetOptions;

it(`uses the valueCaption`, () => {
    option.valueCaption = { key: 'value' };
    initComponent();

    expect(mlFacetValue.valueCaption).toBe(options.valueCaption.key)
})
```

File: unitTests/ui/MLFacet/MLFacetValues/MLFacetValueTest.ts [hunk 2]
```diff
@@ -60,9 +66,37 @@ export function MLFacetValueTest() {
...
+      expect(mLFacetValue.formattedCount).toBe(Globalize.format(mLFacetValue.numberOfResults, 'n0'));
+    });
+
+    it(`when getting formattedCount with a zero value
+      it should return an empty string`, () => {
+      mLFacetValue.numberOfResults = 0;
+      expect(mLFacetValue.formattedCount).toBe('');
+    });
+
+    it(`when using the valueCaption option with a function
```
Comment (samisayegh) [inline unitTests/ui/MLFacet/MLFacetValues/MLFacetValueTest.ts]: Should the test ensures we are passing the original facet value to the function? Atm, we return `allo` all the time.

Maybe something like:

```
{ valueCaption: originalValue => `${originalValue}_caption`}
```

Comment (ThibodeauJF) [reply unitTests/ui/MLFacet/MLFacetValues/MLFacetValueTest.ts]: passing a function simply should not work, not supported

Comment (samisayegh) [reply unitTests/ui/MLFacet/MLFacetValues/MLFacetValueTest.ts]: Oh, I misread and misunderstood the test 👍

**Pull request merged**